src/arc.rs

```diff
@@ -109,15 +109,78 @@ impl Iterator for ArcAppendIter {
     }
 }
 
+/// Take the ellipse radii, how the radii are rotated and the sweep angle, and return a point on
+/// the ellipse.
 fn sample_ellipse(radii: Vec2, x_rotation: f64, angle: f64) -> Vec2 {
     let u = radii.x * angle.cos();
     let v = radii.y * angle.sin();
     rotate_pt(Vec2::new(u, v), x_rotation)
 }
 
+/// Rotate `pt` about the origin by `angle` radians.
 fn rotate_pt(pt: Vec2, angle: f64) -> Vec2 {
     Vec2::new(
         pt.x * angle.cos() - pt.y * angle.sin(),
         pt.x * angle.sin() + pt.y * angle.cos(),
     )
 }
+
+impl Shape for Arc {
+    type BezPathIter = ArcIter;
+
+    fn to_bez_path(&self, tolerance: f64) -> Self::BezPathIter {
```

Comment:
This looks fine to me but you may be able to express the idea more succinctly with something like `std::iter::once(MoveTo(...)).chain(self.append_iter(tolerance))`